src/run_prettify.js

```diff
@@ -398,31 +400,32 @@ var IN_GLOBAL_SCOPE = false;
       // and to defeat aggressive optimizers that fold large string constants.
       var FLOW_CONTROL_KEYWORDS = ["break,continue,do,else,for,if,return,while"];
       var C_KEYWORDS = [FLOW_CONTROL_KEYWORDS,"auto,case,char,const,default," +
-          "double,enum,extern,float,goto,inline,int,long,register,short,signed," +
+          "double,enum,extern,float,goto,inline,int,long,register,restrict,short,signed," +
           "sizeof,static,struct,switch,typedef,union,unsigned,void,volatile"];
       var COMMON_KEYWORDS = [C_KEYWORDS,"catch,class,delete,false,import," +
           "new,operator,private,protected,public,this,throw,true,try,typeof"];
-      var CPP_KEYWORDS = [COMMON_KEYWORDS,"alignof,align_union,asm,axiom,bool," +
+      var CPP_KEYWORDS = [COMMON_KEYWORDS,"alignas,alignof,align_union,asm,axiom,bool," +
           "concept,concept_map,const_cast,constexpr,decltype,delegate," +
           "dynamic_cast,explicit,export,friend,generic,late_check," +
-          "mutable,namespace,nullptr,property,reinterpret_cast,static_assert," +
+          "mutable,namespace,noexcept,noreturn,nullptr,property,reinterpret_cast,static_assert," +
           "static_cast,template,typeid,typename,using,virtual,where"];
       var JAVA_KEYWORDS = [COMMON_KEYWORDS,
           "abstract,assert,boolean,byte,extends,finally,final,implements,import," +
           "instanceof,interface,null,native,package,strictfp,super,synchronized," +
           "throws,transient"];
       var CSHARP_KEYWORDS = [COMMON_KEYWORDS,
-          "abstract,as,async,await,base,bool,by,byte,checked,decimal,delegate,descending," +
-          "dynamic,event,finally,fixed,foreach,from,group,implicit,in,interface," +
-          "internal,into,is,let,lock,null,object,out,override,orderby,params," +
-          "partial,readonly,ref,sbyte,sealed,stackalloc,string,select,uint,ulong," +
-          "unchecked,unsafe,ushort,var,virtual,where"];
+          "abstract,add,alias,as,ascending,async,await,base,bool,by,byte,checked,decimal,delegate,descending," +
```

Comment:
I'm actually re-applying previous patches here that were mistakenly undone later. I didn't add any new C# keywords myself: PR #384 and PR #418